### DISCUSSION TITLE
Integrate with Window list workspace indicator

### DISCUSSION THREAD

**Discussion by wilee8** [mzur/gnome-shell-wsmatrix#206] [Ideas]
I recently switched from Mate to Gnome Shell on my work computer when it was upgraded to RHEL8, and Workspace Matrix has been helpful getting back my preferred Mate workspace configuration in Gnome Shell (6 workspaces in two rows). When I set up this configuration in Mate, the Workspace Switcher indicator in my panel next to the window list would show the workspaces in two rows. I can get the workspace indicator back in Gnome Shell using the [Window List extension](https://extensions.gnome.org/extension/602/window-list/) that is included in Gnome Shell by default, but it always shows the workspaces in one row even if I use Workspace Matrix to put the workspaces in multiple rows. 

I don't know if this is something Workspace Matrix can even do, or if it is something that can only be done inside the Window List extension, but it would be nice if Workspace Matrix provided an option to make the workspace indicator in Window List display the workspaces in the number of rows set in Workspace Matrix if Window List is installed and enabled.

**Comment by wilee8** [comment]
So I just figured out that the only reason my workspace indicator was showing as a horizontal row next to the window list was because I had enabled and disabled [Horizontal workspaces](https://extensions.gnome.org/extension/2141/horizontal-workspaces/) sometime before I turned on Workspace Matrix, normally it's just numbers. So the bad news is that maybe this suggestion should be ignored because it will conflict with Window List *and* Horizontal Workspaces. But the good news is that Horizontal Workspaces shows it it possible to mess with the workspace indicator in Window List.

**Comment by mzur** [comment]
The issue you describe must be addressed by the Window List extension. Workspace Matrix overrides the workspace layout. The other estension probably needs to listen to changes like these and update the workspace display accordingly. However, 2D workspaces are not officially supported by GNOME. So it's unlikely that this will be implemented in the Window List extension (unless you take care of it yourself). Sorry.
